ipaclient/plugins/vault.py

```diff
@@ -691,17 +699,21 @@ def internal(self, algo, *args, **options):
         domain = self.api.env.domain
 
         # try call with cached transport certificate
-        transport_cert = _transport_cert_cache.get(domain)
+        transport_cert = _transport_cert_cache.load_cert(domain)
         if transport_cert is not None:
             result = self._do_internal(algo, transport_cert, False,
                                        *args, **options)
             if result is not None:
                 return result
 
-        # retrieve and cache transport certificate
-        self.api.Command.vaultconfig_show()
-        transport_cert = _transport_cert_cache[domain]
-
+        # retrieve transport certificate
+        response = self.api.Command.vaultconfig_show()
+        transport_cert = x509.load_certificate(
+            response['result']['transport_cert'], x509.DER)
+        # cache it again
+        _transport_cert_cache.store_cert(
+            self.api.env.domain, transport_cert
+        )
```

Comment:
Calling `vaultconfig_show` already writes the cert to the file. There is no reason to do it again here.